### PR TITLE
Skribenten: Unified diff support for word-level letter diffing

### PR DESCRIPTION
**Pull request by routsi**
## Summary

Refactors the word-diff system to support both **split** and **unified** diff formats, and introduces a replace-aware cursor for cleaner diff output.

### Changes

- **New `EditLetterWordTokenizer`**: Extracted tokenization logic into a dedicated class with dual index tracking for insert/delete operations
- **`ReplaceAwareEditScriptCursor`**: Detects when insert+delete operations align at the same position, producing replace entries instead of separate insert/delete
- **`DiffProducer` abstraction**: Builder pattern supporting both split and unified diff output formats
- **API**: `/brev/{brevId}/diff` now defaults to unified diff; pass `split=true` query param for the previous split format

### Testing

- New `EditLetterWordTokenizerTest` covering tokenization edge cases
- New `EditLetterWordUnifiedDiffTest` for unified format output
- Existing tests updated to match refactored structure

### CODE REVIEW

File: skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                Insert(insertPeek.first)
+            }
+            deletePeek?.second != null -> {
+                deleteCursor.requireAndConsume<E>()
+                Delete(deletePeek.first)
+            }
+            insertPeek != null || deletePeek != null -> {
+                if (insertPeek != null) insertCursor.requireAndConsume<E>()
+                if (deletePeek != null) deleteCursor.requireAndConsume<E>()
+                Unchanged((insertPeek ?: deletePeek)!!.first)
```
[the remark this answers is not in the document]
Comment (routsi) [reply skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt]: @copilot This issue should be fixed now. Please review.

File: skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package no.nav.pensjon.brev.skribenten.common.diff
+
+import no.nav.pensjon.brev.skribenten.common.diff.DiffEntry.Delete
+import no.nav.pensjon.brev.skribenten.common.diff.DiffEntry.Insert
+import no.nav.pensjon.brev.skribenten.common.diff.DiffEntry.Replace
+import no.nav.pensjon.brev.skribenten.common.diff.DiffEntry.Unchanged
+
+class ReplaceAwareEditScriptCursor<T : Any>(editScript: EditScript<T>) {
+    @PublishedApi internal val insertCursor = EditScriptCursor(editScript.new, editScript.inserts)
```
Comment (madsop-nav) [inline skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt:9]: Vinn vi noko på @PublishedApi her kontra vanleg public?

**Review by madsop-nav — APPROVED** [review]
Det ser jo fornuftig ut. Vanskeleg å sjå ut frå koden korvidt det vil funke, men testane tydar jo på det.

File: skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt (hunk 2 above)
Comment (routsi) [reply skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/common/diff/ReplaceAwareEditScriptCursor.kt:9]: Siden deklarasjonen er i samme modul som bruken så vinner vi ingenting nei. Det var det beste signalet jeg klarte å få frem om "ikke tukle med cursorene, da ødelegger du tilstanden til den kombinerte cursoren".

**Comment by routsi** [comment]
> Det ser jo fornuftig ut. Vanskeleg å sjå ut frå koden korvidt det vil funke, men testane tydar jo på det.

Det er ikke så veldig lett å se hvorfor det fungerer med den kombinerte cursoren nei, men den vil ihvertfall feile med error om de ikke beveger seg korrekt gjennom brevene som sammenlignes.
